### PR TITLE
Migrate MSVC solution to VS2022

### PR DESCRIPTION
**Pull request by josesimoes**
Description
-----------
- Add configuration to build for Win32 or x64.
- Minor code adjustments to fix compiler warnings.

Test Steps
-----------
- Load solution in Visual Studio. Build.
- Switch to x64 flavour. Build.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
- Discussed in [Forum](https://forums.freertos.org/t/wrong-types-in-x64-build-of-rtos-demo/18715/3).
- For successfully run x64 build, requires code from FreeRTOS/FreeRTOS-Kernel#924.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/freertos_demos.yml [hunk 1]
```diff
@@ -52,17 +52,17 @@ jobs:
         uses: microsoft/setup-msbuild@v1.1
 
       - env:
-          stepName: Build WIN32-MSVC Full Demo
+          stepName: Build WIN32-64-MSVC Full Demo
```
Comment (aggarg) [inline .github/workflows/freertos_demos.yml:55]: Can we avoid renaming the directory and solution file?

File: FreeRTOS/Demo/WIN32-64-MSVC/main.c [hunk 2]
```diff
@@ -80,9 +80,9 @@
  * as this demo could easily create one large heap region instead of multiple
  * smaller heap regions - in which case heap_4.c would be the more appropriate
  * choice.  See http://www.freertos.org/a00111.html for an explanation. */
-#define mainREGION_1_SIZE                     8201
-#define mainREGION_2_SIZE                     23905
-#define mainREGION_3_SIZE                     16807
+#define mainREGION_1_SIZE                     82010
```
Comment (aggarg) [inline FreeRTOS/Demo/WIN32-64-MSVC/main.c:83]: Why is this change needed?

File: .github/workflows/freertos_demos.yml (hunk 1 above)
Comment (josesimoes) [reply .github/workflows/freertos_demos.yml:55]: Sure we can. This change was made to make clear that this can be build for both x86 and x64 arch.
Let me know if you prefer to keep the old name and I'll revert the related changes.

File: FreeRTOS/Demo/WIN32-64-MSVC/main.c (hunk 2 above)
Comment (josesimoes) [reply FreeRTOS/Demo/WIN32-64-MSVC/main.c:83]: When running on x64 there wasn't enough heap. I've simply x10 all of those.

File: .github/workflows/freertos_demos.yml (hunk 1 above)
Comment (aggarg) [reply .github/workflows/freertos_demos.yml:55]: Yes, please.

Comment (josesimoes) [reply .github/workflows/freertos_demos.yml]: Done!

**Review by jasonpcarroll — APPROVED** [review]

**Review by jasonpcarroll — CHANGES_REQUESTED** [review]
Please update manifest file and workflow.

File: .github/workflows/freertos_demos.yml [hunk 3]
```diff
@@ -108,7 +108,7 @@ jobs:
         name: ${{ env.stepName }}
         uses: FreeRTOS/CI-CD-GitHub-Actions/executable-monitor@main
         with:
-            exe-path: FreeRTOS/Demo/WIN32-MSVC/Debug/RTOSDemo.exe
+            exe-path: FreeRTOS/Demo/WIN32-MSVC/x64/Debug/RTOSDemo.exe
```
Comment (Skptak) [inline .github/workflows/freertos_demos.yml:111]: This same change needs to be added to line 80 I believe

**Comment by aggarg** [comment]
This PR required some changes. I cherry-picked your commit and made those changes in a separate PR - https://github.com/FreeRTOS/FreeRTOS/pull/1164.

I am closing this PR in favor of #1164.
